rclcpp/src/rclcpp/node_options.cpp

```diff
@@ -98,7 +98,7 @@ NodeOptions::get_rcl_node_options() const
       }
     }
 
-    if (this->arguments_.size() > std::numeric_limits<int>::max()) {
+    if (this->arguments_.size() > static_cast<size_t>(std::numeric_limits<int>::max())) {
```

Comment:
Wouldn't `std::numeric_limits<size_t>::max()` make more sense?

Reply:
The guard is for the signature of `rcl_parse_arguments`, which accepts an `int` type:

https://github.com/ros2/rcl/blob/b6f4bc97faf34dd85f7acff7cb27945cfe438369/rcl/include/rcl/arguments.h#L91-L98